File: app/src/main/java/com/example/orderfood/adapter/ContactsAdapter.java
package com.example.orderfood.adapter;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.orderfood.R;
import com.example.orderfood.activity.ChatActivity;
import com.example.orderfood.model.User;

import java.util.List;

public class ContactsAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private static final int VIEW_TYPE_HEADER = 0;
    private static final int VIEW_TYPE_CONTACT = 1;

    private final List<Object> items;
    private final OnContactActionsListener actionsListener;

    public ContactsAdapter(List<Object> items, OnContactActionsListener actionsListener) {
        this.items = items;
        this.actionsListener = actionsListener;
    }

    public interface OnContactActionsListener {
        void onEditClick(User contact);
        void onDeleteClick(User contact);
    }

    @Override
    public int getItemViewType(int position) {
        if (items.get(position) instanceof String) {
            return VIEW_TYPE_HEADER;
        } else {
            return VIEW_TYPE_CONTACT;
        }
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        if (viewType == VIEW_TYPE_HEADER) {
            View view = inflater.inflate(R.layout.item_contact_header, parent, false);
            return new HeaderViewHolder(view);
        } else {
            View view = inflater.inflate(R.layout.item_contact, parent, false);
            return new ContactViewHolder(view);
        }
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        int itemViewType = holder.getItemViewType();
        if (itemViewType == VIEW_TYPE_HEADER) {
            HeaderViewHolder headerHolder = (HeaderViewHolder) holder;
            headerHolder.headerTitle.setText((String) items.get(position));
        } else if (itemViewType == VIEW_TYPE_CONTACT) {
            ContactViewHolder contactHolder = (ContactViewHolder) holder;
            User contact = (User) items.get(position);
            contactHolder.contactName.setText(contact.getNickname());

            contactHolder.itemView.setOnClickListener(v -> {
                android.content.Context context = contactHolder.itemView.getContext();
                Intent intent = new Intent(context, ChatActivity.class);
                intent.putExtra("CONTACT_ID", contact.getId());
                intent.putExtra("CONTACT_NICKNAME", contact.getNickname());
                context.startActivity(intent);
            });

            contactHolder.editButton.setOnClickListener(v -> {
                actionsListener.onEditClick(contact);
            });

            contactHolder.deleteButton.setOnClickListener(v -> {
                actionsListener.onDeleteClick(contact);
            });
        }
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public static class ContactViewHolder extends RecyclerView.ViewHolder {
        public final TextView contactName;
        public final Button editButton;
        public final Button deleteButton;

        public ContactViewHolder(@NonNull View itemView) {
            super(itemView);
            contactName = itemView.findViewById(R.id.tv_contact_name);
            editButton = itemView.findViewById(R.id.btn_edit_contact);
            deleteButton = itemView.findViewById(R.id.btn_delete_contact);
        }
    }

    public static class HeaderViewHolder extends RecyclerView.ViewHolder {
        public final TextView headerTitle;

        public HeaderViewHolder(@NonNull View itemView) {
            super(itemView);
            headerTitle = itemView.findViewById(R.id.tv_header);
        }
    }
} 

File: app/src/main/java/com/example/orderfood/adapter/ContactsAdapter.kt
package com.example.orderfood.adapter

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.orderfood.R
import com.example.orderfood.activity.ChatActivity
import com.example.orderfood.model.User

class ContactsAdapter(
    private val items: List<Any>,
    private val actionsListener: OnContactActionsListener
) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    companion object {
        private const val VIEW_TYPE_HEADER = 0
        private const val VIEW_TYPE_CONTACT = 1
    }

    interface OnContactActionsListener {
        fun onEditClick(contact: User)
        fun onDeleteClick(contact: User)
    }

    override fun getItemViewType(position: Int): Int {
        return if (items[position] is String) VIEW_TYPE_HEADER else VIEW_TYPE_CONTACT
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        return if (viewType == VIEW_TYPE_HEADER) {
            val view = inflater.inflate(R.layout.item_contact_header, parent, false)
            HeaderViewHolder(view)
        } else {
            val view = inflater.inflate(R.layout.item_contact, parent, false)
            ContactViewHolder(view)
        }
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        when (holder.itemViewType) {
            VIEW_TYPE_HEADER -> {
                val headerHolder = holder as HeaderViewHolder
                headerHolder.headerTitle.text = items[position] as String
            }
            VIEW_TYPE_CONTACT -> {
                val contactHolder = holder as ContactViewHolder
                val contact = items[position] as User
                contactHolder.contactName.text = contact.nickname

                contactHolder.itemView.setOnClickListener {
                    val context = contactHolder.itemView.context
                    val intent = Intent(context, ChatActivity::class.java).apply {
                        putExtra("CONTACT_ID", contact.id)
                        putExtra("CONTACT_NICKNAME", contact.nickname)
                    }
                    context.startActivity(intent)
                }

                contactHolder.editButton.setOnClickListener {
                    actionsListener.onEditClick(contact)
                }

                contactHolder.deleteButton.setOnClickListener {
                    actionsListener.onDeleteClick(contact)
                }
            }
        }
    }

    override fun getItemCount(): Int = items.size

    class ContactViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val contactName: TextView = itemView.findViewById(R.id.tv_contact_name)
        val editButton: Button = itemView.findViewById(R.id.btn_edit_contact)
        val deleteButton: Button = itemView.findViewById(R.id.btn_delete_contact)
    }

    class HeaderViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val headerTitle: TextView = itemView.findViewById(R.id.tv_header)
    }
} 